src/Microsoft.ML.TorchSharp/Utils/TorchUtils.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using TorchSharp;
+
+namespace Microsoft.ML.TorchSharp.Utils
+{
+    internal static class TorchUtils
+    {
+        public static void DisposeDictionaryWithTensor<TKey, TResult>(Dictionary<TKey, TResult> dictionary)
+        {
+            if (dictionary == null)
+            {
+                return;
+            }
+
+            foreach (var value in dictionary.Values)
```

Comment:
>f [](http://example.com/codeflow?start=12&length=1)

why you don't enumerate both keys and values in same time? 

foreach (var kvp in dictionary)